framework/db/plugin_uploader.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import print_function
+from sys import getsizeof, stderr
+from itertools import chain
+from collections import deque
+import os
+from datetime import datetime
+import re
+
+from ptp import PTP
+from framework.dependency_management.dependency_resolver import ServiceLocator
+from framework.db import models
+from framework.db.target_manager import target_required
+
+class PluginUploader():
+
+	def __init__(self, tool_name):
+		self.supported_tools = ['w3af', 'skipfish', 'arachni', 'hoppy', 'burpsuite']
+		#self.full_parse_tools = ['w3af', 'skipfish', 'arachni']
+		self.db = ServiceLocator.get_component("db")
+		self.transaction = ServiceLocator.get_component("transaction")
+		self.re_extras = re.compile(r"(.*?) (\/.*?||\w+.*?) HT.*?")
+		if tool_name in self.supported_tools:
+			self.tool_name = tool_name
+			self.ptp = PTP(tool_name)
```

Comment:
Please consider catching all PTP exception like in https://github.com/owtf/owtf/blob/develop/framework/plugin/plugin_handler.py#L279

If the Plugin Uploader is not mandatory, consider doing the same as in https://github.com/owtf/owtf/blob/develop/framework/plugin/plugin_handler.py#L281